site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+            </ul>
+        </li>
+        <li>Keep it short.
+            <ul>
+                <li>Preferably use less than five words.</li>
+                <li>If the user needs more information for filling the information, it <em>should</em> be provided with assistive text or tooltip.</li>
+            </ul>
+        </li>
+    </ul>
+</div>
+
+<div class="guideline-dont" style="background: var(--color-error-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-dont-label" style="color:var(--color-error); margin-bottom:var(--spacing-s);"><IconError style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Don't</strong></div>
+    <ul style="margin-bottom:0;">
+        <li>Use ALL CAPS.</li>
+        <li>Use a colon : after label names.</li>
+    </ul>
+</div>
+
+---
+
+### 3. Required field indicator
+
+Marking that indicates a form control for which **the user *must* enter a value** in order to successfully submit the form. 
+
+<Image src="../../static/patterns/form/form-elements-required@2x.png" alt="Form control required indicator" style="max-width:376px;" viewable /> 
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li><strong>aria-required="true" attribute <em>must</em> be always provided for required fields</strong>, even if they are not marked visually.</li>
+        <li>If all fields in the form are required
+            <ul>
+                <li>This <em>should</em> be stated in the form wide instructions at the beginning of the form.</li>
+                <li>The submit-button can also be disabled before all fields are filled.</li>
+            </ul>
+        </li>
+    </ul>
+</div>
+
+---
+
+### 4. Icon
+
+Icon is a **supplementary visual representation of the label** displayed inside the form field.
+
+Icons can be used to:
+- **Emphasise** important form controls visually.
+- **Differentiate** similar form controls easier from each other.
+
+<Image src="../../static/patterns/form/form-elements-icon@2x.png" alt="Form control icon" style="max-width:376px;" viewable />
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li>Be descriptive.
```

Comment:
> Be descriptive.
Choose an icon that clearly describes the type fo input or purpose of the form control
Do not use arbitrary or vague icons just for visual purposes.

Myös näissä listoissa on hieman epäjohdonmukaisesti käytetty pistettä osan bullettien perässä ja osassa ei.